### PR TITLE
docs: point out throw caveat

### LINKED ISSUE #8689: Clarify and Document use of 'throw redirect()' in try-catch block
### Describe the problem

I have been wrecking my brain for weeks trying to figure out why my redirects where not working. I finally discovered it was because the `throw redirect()` was inside a try-catch block, see example below:
```js
export const actions : {
	default: async(event) => {
		
		try {
			authenticate(...);
			throw redirect(301, '/success');
		} catch (error) {
			// Redirect being caught here as an error
			return fail()
		}
	
};
```
The `redirect()` is being caught as an error and the route won't redirect. I encountered this problem while migrating code from an older sveltekit project where the try-catch pattern of coding worked perfectly by returning a header instead. It wasn't obvious to me what I was doing wrong and console logging the `error` did not provide any clues to that end. I think I missed it, and many more might, because of Sveltekit's innovative use of `throw`.

### Describe the proposed solution

1. I suggest putting a warning in the documentation, especially for the sake of beginners, so that if anyone get's stuck and turn to the docs, they can find a clue.

2. Sveltekit displays a more descriptive error in the console if the try-catch or similar pattern is to be used in conjunction with `throw redirect()` and similar `throws`.

### Alternatives considered

_No response_

### Importance

would make my life easier

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by dummdidumm**
closes #8689

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [ ] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [ ] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### CODE REVIEW

**Review by Conduitry — COMMENTED** [review]
I would suggest instead saying something like "make sure you're not catching the exception yourself without re-throwing it", which is a bit more accurate.

**Comment by geoffrich** [comment]
Is it possible to document how to tell that you caught a Kit redirect as opposed to an exception? e.g. an instanceof check or something

File: packages/kit/types/index.d.ts [hunk 1]
```diff
@@ -1110,6 +1111,8 @@ export interface HttpError {
 
 /**
  * Create a `Redirect` object. If thrown during request handling, SvelteKit will return a redirect response.
+ * Make sure you're not catching the thrown redirect, which results in a noop.
+ *
```
Comment (benmccann) [inline packages/kit/types/index.d.ts:1115]: We mostly don't have a blank line before the `@param` tags and you're not adding one above, so probably better to leave it out for consistency
```suggestion
```

**Comment by dummdidumm** [comment]
> Is it possible to document how to tell that you caught a Kit redirect as opposed to an exception? e.g. an instanceof check or something

honestly, I don't think we have a way to do that right now, but also noone expressed the desire to do so yet, so I would let that slide for now

**Review by Rich-Harris — APPROVED** [review]

**Comment by geoffrich** [comment]
> honestly, I don't think we have a way to do that right now, but also noone expressed the desire to do so yet, so I would let that slide for now

This was requested in #7941, but looks like that issue was closed too soon - I'll re-open.

**Pull request merged**